lxd/instance/drivers/driver_qemu.go

```diff
@@ -1571,6 +1571,40 @@ func (d *qemu) Start(stateful bool) error {
 	return nil
 }
 
+// advertiseVsockAddress advertises the CID and port to the VM.
+func (d *qemu) advertiseVsockAddress() error {
+	devlxdEnabled := d.expandedConfig["security.devlxd"] == "" || shared.IsTrue(d.expandedConfig["security.devlxd"])
+
+	client, err := d.getAgentClient()
+	if err != nil {
+		return fmt.Errorf("Failed getting agent client handle: %w", err)
+	}
+
+	agent, err := lxd.ConnectLXDHTTP(nil, client)
+	if err != nil {
+		return fmt.Errorf("Failed connecting to lxd-agent: %w", err)
+	}
+
+	defer agent.Disconnect()
+
+	var req api.API10Put
```

Comment:
Any specific reason for not doing it in one shot?

```go
req := api.API10Put{
    Certificate: string(d.state.Endpoints.NetworkCert().PublicKey()),
    Devlxd: devlxdEnabled
}
```

Reply:
No, no specific reason.